### PR TITLE
Add type definitions for ari-client

### PR DESCRIPTION
**Pull request by dioris-moreno**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [X] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/ari-client/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,2003 @@
+// Type definitions for ari-client 2.2
+// Project: https://github.com/asterisk/node-ari-client
+// Definitions by: Dioris Moreno <https://github.com/dioris-moreno>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+// tslint:disable-next-line:export-just-namespace
+export = Ari;
+export as namespace Ari;
```
Comment (armanio123) [inline types/ari-client/index.d.ts:10]: I don't think this is an UMD module, so this line looks wrong. 

I'm not very familiar with the library. Just double check this is the case and if you feel is alright I can help you with the merge.

**Review by armanio123 — CHANGES_REQUESTED** [review]

File: types/ari-client/index.d.ts (hunk 1 above)
Comment (dioris-moreno) [reply types/ari-client/index.d.ts]: Asterisk is an open source framework for building communications applications, and "ari-client" is the Node.js library for the Asterisk RESTful Interface (ARI), an asynchronous API that allows developers to build those applications. 

You are right, this is not an UMD library. I rewritten the declaration based on the module-class template (https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-class-d-ts.html) as follows:

export = Ari;

declare namespace Ari {
    function connect(
        baseUrl: string,
        user: string,
        pass: string,
        callback?: (err: Error, client: Client) => void,
    ): Promise<Client>;

    interface Client extends Resource {
    ...

The idea of exposing types and interfaces in Ari namespace is to allow destructured import of Asterisk objects (Channel, Bridge, Endpoint, etc.). Developers will need to turn on esModuleInterop in order to use an import statement like this:

import Ari, { Channel, Bridge, Endpoint} from 'ari-client';

Comment (dioris-moreno) [reply types/ari-client/index.d.ts]: I just pushed the updated version.

Comment (armanio123) [reply types/ari-client/index.d.ts]: I don't think the namespace is necessary. I'm inclined to let the tslint rule on and have every interface exported instead. 

If you want to have a syntax like `import Ari, { Channel, Bridge, Endpoint} from 'ari-client';`, all you need to do is enable the `esModuleInterop` as you mentioned.

Comment (dioris-moreno) [reply types/ari-client/index.d.ts]: I just pushed the changes you suggested. It is much better.

**Review by armanio123 — APPROVED** [review]

**Pull request merged**